lewis/adapters/stream.py

```diff
@@ -53,20 +53,21 @@ def found_terminator(self):
 
         self.log.debug('Got request %s', request)
 
-        try:
-            cmd = next((cmd for cmd in self.target.bound_commands if cmd.can_process(request)),
-                       None)
+        with self._stream_server.device_lock:
```

Comment:
No looping issues here... TCP stream commands always came one-by-one, without state-coherence guarantee.

This also matches real device behaviour as far as I understand. That's why devices like the Linkam T95 have commands like `Get T Status`, which returns all state information in one go.